cli/create.go

```diff
@@ -170,10 +170,40 @@ func create() *cobra.Command {
 			}
 			_, _ = fmt.Fprintln(cmd.OutOrStdout())
 
-			resources, err := client.TemplateVersionResources(cmd.Context(), templateVersion.ID)
+			// Run a plan with the given parameters to check correctness
+			after := time.Now()
+			planJob, err := client.CreateTemplateVersionPlan(cmd.Context(), templateVersion.ID, codersdk.CreateTemplateVersionPlanRequest{
+				WorkspaceName:   workspaceName,
+				ParameterValues: parameters,
+			})
 			if err != nil {
-				return err
+				return xerrors.Errorf("begin workspace plan: %w", err)
+			}
+			_, _ = fmt.Fprintln(cmd.OutOrStdout(), "Planning workspace...")
+			err = cliui.ProvisionerJob(cmd.Context(), cmd.OutOrStdout(), cliui.ProvisionerJobOptions{
+				Fetch: func() (codersdk.ProvisionerJob, error) {
+					return client.TemplateVersionPlan(cmd.Context(), templateVersion.ID, planJob.ID)
+				},
+				Cancel: func() error {
+					return client.CancelTemplateVersionPlan(cmd.Context(), templateVersion.ID, planJob.ID)
+				},
+				Logs: func() (<-chan codersdk.ProvisionerJobLog, error) {
+					return client.TemplateVersionPlanLogsAfter(cmd.Context(), templateVersion.ID, planJob.ID, after)
+				},
+				// Don't show log output for the plan unless there's an error.
+				Silent: true,
+			})
+			if err != nil {
+				// TODO (Dean): reprompt for parameter values if we deem it to
+				// be a validation error
```

Comment:
TODO: open an issue

Reply:
Opened two issues:

https://github.com/coder/coder/issues/1872 - for integrating with frontend
https://github.com/coder/coder/issues/1873 - for reprompting for values in CLI